gpu/gpu.py

```diff
@@ -0,0 +1,65 @@
+from random import random
+
+
+def check_squared_value(x, xSquer, testValue):
+    if x == testValue - 1:
+        if x != testValue - 1:
+            return False
+        else:
+            return True
+    if xSquer == 1:
+        return False
+
+
+# Run function on gpu "repetitions" cores
+def testing_value(testValue, testValueMinusOne, powerOfTwo):
+    # Choose a random number between 2 and testValue - 2
+    randomValue = int(random() * (testValue - 3)) + 2
+
+    # Calculate randomValue^testValueMinusOne mod testValue
+    x = pow(randomValue, testValueMinusOne, testValue)
+
+    # If x is 1 or testValue - 1, we don't know if testValue is prime, so we'll try again
+    if x == 1 or x == testValue - 1:
+        return True
+
+    # For "powerOfTwo" gpu cores:
+    for _ in range(1, powerOfTwo):
+
+        # Divide between gpu cores?
+        #
+        xSquer = pow(x, 2, testValue)
+        check_squared_value(x, xSquer, testValue)
+        x = xSquer
+        #
+
+    # if "all cores return True":
+    #     return True
+    # else:
+    #     return False
```

Comment:
Tu logika cos się trochę rozjezdza ale powiedzmy że to do dorobienia już w trakcie konkretnego kodowania na GPU